### ISSUE TITLE
Request: Separate comma and semicolon checks in whitespace's check-separator

### ISSUE THREAD

**Issue by Jessidhia** [palantir/tslint#2929]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.4.0
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
;(foo as bar).baz()
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "whitespace": [true, "check-separator"]
  }
}
```

#### Actual behavior

Error is raised requiring a space between the `;` and the opening `(`

#### Expected behavior

A way of separating the enforcement of space after a comma (`, `) from the enforcement of having a space after a semicolon. This shows up in semicolon-free styles when a line needs to start with a `(` or `[`.

See https://github.com/blakeembrey/tslint-config-standard/pull/30#issuecomment-308528238

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting formatting/whitespace requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!
